packages/component-library/components/Heading/Heading.elm

```diff
@@ -0,0 +1,194 @@
+    )
+
+import CssModules exposing (css)
+import Debug exposing (log)
+import Html exposing (text)
+import Html.Attributes
+
+
+type alias ConfigValue msg =
+    { tag : Element msg
+    , variant : TypeVariant
+    , id : Maybe String
+    }
+
+
+type BaseConfig msg
+    = BaseConfig (ConfigValue msg)
+
+
+type Config msg
+    = Config (ConfigValue msg)
+
+
+type alias Element msg =
+    List (Html.Attribute msg) -> List (Html.Html msg) -> Html.Html msg
+
+
+defaultConfig : ConfigValue msg
+defaultConfig =
+    { tag = Html.div
+    , id = Nothing
+    , variant = Heading1
+    }
+
+
+
+-- VIEW
+
+
+view : Config msg -> List (Html.Html msg) -> Html.Html msg
+view (Config config) children =
+    let
+        resolveId =
+            case config.id of
+                Just id_ ->
+                    [ Html.Attributes.id id_ ]
+
+                Nothing ->
+                    []
+    in
+    config.tag
+        ([ className config.variant ] ++ resolveId)
+        children
+
+
+className : TypeVariant -> Html.Attribute msg
+className typeVariant =
+    let
+        styleClass =
+            case typeVariant of
+                Display0 ->
+                    .display0
+
+                Heading1 ->
+                    .heading1
+
+                Heading2 ->
+                    .heading2
+
+                Heading3 ->
+                    .heading3
+
+                Heading4 ->
+                    .heading4
+
+                Heading5 ->
+                    .heading5
+
+                Heading6 ->
+                    .heading6
+    in
+    styles.classList
+        [ ( styleClass, True )
+        , ( .heading, True )
+        ]
+
+
+styles =
+    css "@kaizen/component-library/components/Heading/Heading.module.scss"
+        { heading = "heading"
+        , display0 = "display-0"
+        , heading1 = "heading-1"
+        , heading2 = "heading-2"
+        , heading3 = "heading-3"
+        , heading4 = "heading-4"
+        , heading5 = "heading-5"
+        , heading6 = "heading-6"
+        }
+
+
+type TypeVariant
+    = Display0
+    | Heading1
+    | Heading2
+    | Heading3
+    | Heading4
+    | Heading5
+    | Heading6
+
+
+
+-- MODIFIERS
+
+
+pre : Config msg
+pre =
+    Config { defaultConfig | tag = Html.pre }
+
+
+p : Config msg
+p =
+    Config { defaultConfig | tag = Html.p }
+
+
+a : Config msg
+a =
+    Config { defaultConfig | tag = Html.a }
+
+
+div : Config msg
+div =
+    Config { defaultConfig | tag = Html.div }
+
+
+span : Config msg
+span =
+    Config { defaultConfig | tag = Html.span }
+
+
+h1 : Config msg
+h1 =
+    Config { defaultConfig | tag = Html.h1 }
+
+
+h2 : Config msg
+h2 =
+    Config { defaultConfig | tag = Html.h2 }
+
+
+h3 : Config msg
+h3 =
+    Config { defaultConfig | tag = Html.h3 }
+
+
+h4 : Config msg
+h4 =
+    Config { defaultConfig | tag = Html.h4 }
+
+
+h5 : Config msg
+h5 =
+    Config { defaultConfig | tag = Html.h5 }
+
+
+h6 : Config msg
+h6 =
+    Config { defaultConfig | tag = Html.h6 }
```

Comment:
Looks like if you forget to explicitly set a variant modifier `|> variant Heading6` then it will default to `Heading1` which seems like a strange default for a `h6`. Is this what we want? Is this also how the React component works?

Reply:
I think this is now at feature parity with the React component. The only required prop is `variant`. If `tag` is not supplied, the component "guesses" the best tag match based on what the variant is in `inferTagFromVariant` (heading-1 will output h1, heading-2 will output h2, etc).

I've added `classNameAndIHaveSpokenToDST`. I notice that the custom classnames don't get hashed, so there might be some Elm wizardry that I don't quite understand yet.

Reply:
Oh, the classname hashing issue was for the `classNameAndIHaveSpokenToDST` parameter. The only answer to this I think is for the consumer to provide the already hashed string. There's an Elm function to do this in the Elm CSS Modules library that Kevin built, but people might not know about it.... but _they will speak to DST_, and we can tell them when they ask! (that name keeps on giving!) It's the `toString` function here: https://package.elm-lang.org/packages/cultureamp/elm-css-modules-loader/latest/CssModules#Helpers.

Reply:
I wrote this a little prematurely - let me think about it some more.

Reply:
I'll update the Story to use CSSModules `toString` function and it should work.

Reply:
I added a crazy blinking text example in the Story: 
<img width="283" alt="image" src="https://user-images.githubusercontent.com/702885/76283702-6cc86680-62ef-11ea-9188-6c1bf5966021.png">